core/src/main/scala/com/yahoo/maha/core/query/QueryContext.scala

```diff
@@ -114,7 +114,10 @@ case class DimensionBundle(dim: Dimension
                            , hasLowCardinalityFilter: Boolean
                             ) {
   //only filtering on primary key alias then subquery candidate
-  lazy val isSubQueryCandidate: Boolean = fields.filterNot(publicDim.isPrimaryKeyAlias).isEmpty
+  lazy val filterFields: SortedSet[String] = for (filter <- filters) yield {
+    filter.field
+  }
+  lazy val isSubQueryCandidate: Boolean = (fields ++ filterFields).forall(publicDim.isPrimaryKeyAlias)
```

Comment:
equivalent to (fields ++ filterFields).filterNot(publicDim.isPrimaryKeyAlias).isEmpty